datafusion/sql/src/relation/mod.rs

```diff
@@ -299,6 +300,108 @@ impl<S: ContextProvider> SqlToRel<'_, S> {
                     }
                 }
             }
+            TableFactor::Unpivot {
+                table,
+                null_inclusion,
+                value,
+                name,
+                columns,
+                alias,
+            } => {
+                let base_plan = self.create_relation(*table, planner_context)?;
+                let base_schema = base_plan.schema();
+
+                let value_column = value.value.clone();
+                let name_column = name.value.clone();
+
+                let mut unpivot_column_indices = Vec::new();
+                let mut unpivot_column_names = Vec::new();
+
+                let mut common_type = None;
+
+                for column_ident in &columns {
+                    let column_name = column_ident.value.clone();
+
+                    if let Some(idx) =
```

Comment:
nitpick: I personally find it easier to read when with an "exit early" approach, subjectively this can be rewritten as 
```rust
    let idx = if let Some(i) = base_schema.index_of_column_by_name(None, &column_name) {
        i
    } else {
        return plan_err!("Column '{}' not found in input", column_name);
    };

    let field = base_schema.field(idx);
    let field_type = field.data_type();
```